packages/plugin-core/src/features/ReferenceHoverProvider.ts

```diff
@@ -1,122 +1,115 @@
-          if (containsImageExt(foundUri.fsPath)) {
-            return `![${
-              isUncPath(foundUri.fsPath)
-                ? "UNC paths are not supported for images preview due to VSCode Content Security Policy. Use markdown preview or open image via cmd (ctrl) + click instead."
-                : ""
-            }](${vscode.Uri.file(
-              foundUri.fsPath
-            ).toString()}|height=${imageMaxHeight})`;
-          } else if (containsOtherKnownExts(foundUri.fsPath)) {
-            const ext = path.parse(foundUri.fsPath).ext;
-            return `Preview is not supported for "${ext}" file type. Click to open in the default app.`;
-          }
+  /** Returns a message if this is a non-dendron URI. */
+  private handleNonDendronUri(refText: string): string | undefined {
+    const [first, second] = refText.split("|");
+    const maybeUri = second || first;
+    const maybe = containsNonDendronUri(maybeUri);
+    // Not a URI, or is dendron://, so it must be a note (or image) and the rest of the code can handle this.
+    if (_.isUndefined(maybe) || !maybe) return undefined;
+    // Otherwise, this is a URI like http://example.com or mailto:user@example.com
+    return `Preview is not supported for this link. [Click to open in the default app.](${maybeUri}).`;
+  }
 
-          const fname = path.basename(foundUri.fsPath, ".md");
-          const vault = PickerUtilsV2.getOrPromptVaultForOpenEditor();
-          const note = NoteUtils.getNoteByFnameV5({
-            fname,
-            vault,
-            notes: getWS().getEngine().notes,
-            wsRoot: DendronWorkspace.wsRoot(),
-          });
-          if (!note) {
-            throw new DendronError({ message: `note ${fname} not found` });
-          }
-          const out = await new MarkdownPublishPod().plant({
-            note,
-            config: {
-              fname,
-              dest: "stdout",
-              vault: PickerUtilsV2.getOrPromptVaultForOpenEditor(),
-            },
-            engine: getWS().getEngine(),
-            vaults: getWS().vaultsv4,
-            wsRoot: DendronWorkspace.wsRoot(),
-          });
+  public async provideHover(
+    document: vscode.TextDocument,
+    position: vscode.Position
+  ): Promise<vscode.Hover | null> {
+    const refAtPos = getReferenceAtPosition(document, position);
+    if (!refAtPos) return null;
+    const { range } = refAtPos;
+    const hoverRange = new vscode.Range(
+      new vscode.Position(range.start.line, range.start.character + 2),
+      new vscode.Position(range.end.line, range.end.character - 2)
+    );
 
-          return out;
-        };
-        const content = await getContent();
+    const engine = getEngine();
+    const vault = PickerUtilsV2.getOrPromptVaultForOpenEditor();
 
-        return new vscode.Hover(content, hoverRange);
-      }
+    // Check if what's being referenced is a note.
+    const maybeNotes = NoteUtils.getNotesByFname({
```

Comment:
we could have a xvault link in which case the vault also needs to be passed in. that would be `ref.vaultName` property